Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP_test.py

```diff
@@ -0,0 +1,1018 @@
+"""
+Unit testing for CiscoAMP (Advanced Malware Protection)
+"""
+import json
+import io
+import os
+from typing import Dict, List, Any
+import pytest
+from CiscoAMP import Client
+
+API_KEY = 'API_Key'
+CLIENT_ID = 'Client_ID'
+SERVER_URL = 'https://api.eu.amp.cisco.com'
+BASE_URL = f'{SERVER_URL}/{Client.API_VERSION}'
+
+
+def assert_output_has_no_links(outputs: List[Dict]):
+    """
+    Check that there are no 'links' keys in the outputs.
+
+    Args:
+        outputs (List[Dict, str]): output to loop through.
+    """
+    for output in outputs:
+        assert 'links' not in output
+
+
+def load_mock_response(file_name: str) -> str | io.TextIOWrapper:
+    """
+    Load mock file that simulates an API response.
+    Args:
+        file_name (str): Name of the mock response JSON file to return.
+    Returns:
+        str: Mock file content.
+    """
+    path = os.path.join('test_data', file_name)
+
+    with io.open(path, mode='r', encoding='utf-8') as mock_file:
+        if os.path.splitext(file_name)[1] == '.json':
+            return json.loads(mock_file.read())
+
+        return mock_file
+
+
+@pytest.fixture(autouse=True)
+def mock_client() -> Client:
+    """
+    Establish a connection to the client with a URL and API key.
+
+    Returns:
+        Client: Connection to client.
+    """
+    return Client(
+        server_url=SERVER_URL,
+        api_key=API_KEY,
+        client_id=CLIENT_ID,
+    )
+
+
+@pytest.mark.parametrize(
+    'args, suffix, file',
+    [({'limit': '34'}, '', 'computer_list_response.json'),
+     ({'connector_guid': '1'}, '/1', 'computer_get_response.json'),
+     ({'connector_guid': '1', 'group_guid': '2'}, '/1', 'computer_get_response.json')]
+)
+def test_computer_list_command(requests_mock, mock_client, args, suffix, file):
+    """
+    Scenario:
+    -   Get a list of 34 computers.
+    -   Get a single computer.
+    -   Search for a specific computer and get a list of computers in a group.
+    Given:
+    -   The user has entered a limit.
+    -   The user has entered a connector_guid.
+    -   The user has entered a connector_guid and a group_guid.
+    When:
+    -    cisco-amp-computer-list is called.
+    Then:
+    -   Ensure outputs_prefix is correct.
+    -   Ensure links don't exist.
+    OR
+    -   Ensure an exception has been raised.
+    """
+    mock_response = load_mock_response(file)
+    requests_mock.get(
+        f'{BASE_URL}/computers{suffix}',
+        json=mock_response
+    )
+
+    try:
+        from CiscoAMP import computer_list_command
+        responses = computer_list_command(mock_client, args)
+
+        for response in responses[:-1]:
+            assert response.outputs_prefix == 'CiscoAMP.Computer'
+            assert 'links' not in response.outputs
+
+    except ValueError as ve:
+        assert str(ve) == 'connector_guid must be the only input, when fetching a specific computer.'
```

Comment:
please separate this to a different test, as this prevents checking all the values (if the first value raises, the test passes and ends, even if the later are buggy)

This comment applies to all tests in this file. The standard is to have a test with `with pytest.raises(ValueError) as e`, and a test for the "happy path" where things do work as expected.

Reply:
Thanks for teaching me some thing new! 🙂